generators/app/index.js

```diff
@@ -15,9 +15,23 @@ const packageJsonFile = 'package.json';
 class NgxGenerator extends Generator {
   initializing() {
     this.version = pkg.version;
-    this.insight = new Insight({trackingCode: 'UA-93069862-1', pkg});
     this.props = {};
 
+    // Try to initialize analytics. Insight is broken for some users, so if it fails, proceed as if the --no-analytics flag was present.
+    try {
+      this.insight =
+        !this.options.analytics || process.env.DISABLE_NGX_ANALYTICS
```

Comment:
Factorize this with the definition of `insight.optOut` (currently declared on line 48).